File: game/scripts/vscripts/legacy/items/item_health_orb/item_health_orb.lua
item_health_orb = class({})

function item_health_orb:OnSpellStart()
    local caster = self:GetCaster()
    local heal = 3

    caster:Heal(heal, caster)
    CustomEntitiesLegacy:TrueHeal(caster, heal)

    EmitSoundOn("DOTA_Item.FaerieSpark.Activate", caster)
end

File: src/vscripts/items/item_health_orb.ts
import { CustomItem } from "../abilities/framework/custom_ability";
import { registerAbility } from "../lib/dota_ts_adapter";

@registerAbility("item_health_orb")
class ItemHealthOrb extends CustomItem {
    OnSpellStart() {
        const heal = 3;

        this.GetCaster().Heal(heal, this);
        CustomEntitiesLegacy.TrueHeal(this.GetCaster(), heal);

        EmitSoundOn("DOTA_Item.FaerieSpark.Activate", this.GetCaster());
    }
}
